JPEGImageCompression/image_compression.py

```diff
@@ -1,19 +1,145 @@
+        """
+        max_color_res = 256
+        color_resolution = color_res
+        discrete_ratio = max_color_res // color_resolution
+
+        side_length = cube_side_length
+        offset = side_length + buff
+        cubes = []
+
+        for i in range(color_resolution):
+            for j in range(color_resolution):
+                for k in range(color_resolution):
+
+                    color = rgb_to_hex(
+                        (
+                            (i * discrete_ratio) / 255,
+                            (j * discrete_ratio) / 255,
+                            (k * discrete_ratio) / 255,
+                        )
+                    )
+                    cubes.append(
+                        Cube(
+                            side_length=side_length, fill_color=color, fill_opacity=1
+                        ).shift((LEFT * i + UP * j + OUT * k) * offset)
+                    )
+
+        cubes_vg = Group(*cubes)
+        return cubes_vg
+
+    def create_yuv_cube(self, color_res=8, cube_side_length=0.1, buff=0.05):
+        """
+        Creates a YCbCr cube composed of many smaller cubes. The `color_res` argument defines
+        how many cubes there will be to represent the full spectrum, particularly color_res ^ 3.
+        Works exactly like `create_rgb_cube` but for YCrCb colors.
+
+        It is recommended that color_res is a power of two.
+
+        @param: color_res - defines the number of cubes in every dimension. Higher values yield a finer
+        representation of the space, but are very slow to deal with.
+        @param: cube_side_length - defines the side length of each individual cube
+        @param: buff - defines how much space there will be between each cube
+        """
+
+        max_color_res = 256
+        color_resolution = color_res
+        discrete_ratio = max_color_res // color_resolution
+
+        side_length = cube_side_length
+        offset = side_length + buff
+        cubes = []
+
+        for i in range(color_resolution):
+            for j in range(color_resolution):
+                for k in range(color_resolution):
+
+                    r = i * discrete_ratio
+                    g = j * discrete_ratio
+                    b = k * discrete_ratio
+
+                    y, cb, cr = self.rgb2ycbcr(r, g, b)
+
+                    color = rgb_to_hex(
+                        (
+                            (y) / 255,
+                            (cb) / 255,
+                            (cr) / 255,
+                        )
+                    )
+
+                    cubes.append(
+                        Cube(
+                            side_length=side_length, fill_color=color, fill_opacity=1
+                        ).shift((LEFT * i + UP * j + OUT * k) * offset)
+                    )
+
+        cubes_vg = Group(*cubes)
+        return cubes_vg
+
+    def rgb2ycbcr(self, r, g, b):  # in (0,255) range
```

Comment:
I recommend making these functions that we define outside the class, since I imagine we may need these computations in other animations down the line. Define them similar to how I defined `gray_scale_to_hex`